File: packages/contracts/src/types/match.ts
export interface MatchExplanation {
  skills: { matched: string[]; required: string[]; ratio: number };
  title: { overlap: string[]; job_tokens: string[]; ratio: number };
  experience: { overlap: string[]; ratio: number };
  education: { matched: string[]; ratio: number };
  weights: { skills: number; title: number; experience: number; education: number };
}

export interface JobMatch {
  id: string;
  jobId: string;
  candidateProfileId: string;
  profileVersion: number;
  inputFingerprint: string;
  eligible: boolean;
  score: number;
  reasons: string[];
  explanation: MatchExplanation;
  createdAt: string | null;
}

File: services/api/app/schemas/match.py
from __future__ import annotations

from datetime import datetime
from typing import Any

from pydantic import BaseModel, Field

from app.matching.engine import MatchConstraints


class MatchSchema(BaseModel):
    id: str
    job_id: str
    candidate_profile_id: str
    profile_version: int
    input_fingerprint: str
    eligible: bool
    score: float
    reasons: list[str] = Field(default_factory=list)
    explanation: dict[str, Any]
    created_at: datetime | None = None

    model_config = {"from_attributes": True}


class MatchRefreshSchema(BaseModel):
    profile_id: str
    profile_version: int | None = Field(default=None, ge=1)
    job_ids: list[str] | None = None
    constraints: MatchConstraints = Field(default_factory=MatchConstraints)


class MatchRefreshResponse(BaseModel):
    created: int
    existing: int
    matches: list[MatchSchema]


class ProfileRefreshSummary(BaseModel):
    profile_id: str
    profile_version: int
    created: int
    existing: int


class MatchRefreshAllResponse(BaseModel):
    """Summary of a match refresh across every candidate profile."""

    profiles: int
    jobs: int
    created: int
    existing: int
    per_profile: list[ProfileRefreshSummary]
